### ISSUE TITLE
F-Droid: Geolocation

### ISSUE THREAD

**Issue by newhinton** [home-assistant/android#1511]
**Is your feature request related to a problem? Please describe.**
Geolocation is not available for F-Droid. See #1335 

**Describe the solution you'd like**
I found a newer library: Its last commit was from late 2020, so it seems active enough. Also, the implementation seems rather straitforward, implementing a sample application took less than 10 minutes.

**Describe alternatives you've considered, if any**
See #1335 

**Additional context**

**Comment by dshokouhi** [comment]
Nice find! Will need to research this a bit.  Also this won't just benefit the minimal flavor if it works we will use it for all flavors!

**Comment by gdt** [comment]
My impression is that UnifiedNLP is available on all the ROMs that don't have gmscore, and that it's straightforward to use.  The only issue is not insisting that the provider of location services be signed by Google; on many ROMs there isn't "signature spoofing".   As an example, OsmAnd (navigation) uses GPS and "network location" just fine on Lineage.   In my case "network" is a plugin called DejaVu, that keeps on-device records of wifi/cell locations, basically like querying a network database except it's local.

**Comment by newhinton** [comment]
Id like to keep this open as this creates feature parity between stores.

**Comment by bd4** [comment]
I just ran into this with latest stable HA and app from f-droid, on CalyxOS Pixel 3a. Would be great to have location support on de-googled phones like Calyx and /e/.

**Comment by d-two** [comment]
Same here... My wife has a Huawei Honor 10 X without google apps and all stuff work except the Location :-(

**Comment by mike-lloyd03** [comment]
Is it possible to implement something like [UnifiedNlp](https://github.com/microg/UnifiedNlp) or Mozilla's [MLS](https://location.services.mozilla.com/) on the minimal version?

**Comment by gdt** [comment]
Basically it needs to use the AOSP location API, which has both GPS and network, instead of using the Play Services API.   Google's docs are buggy in that they tell people to use the Play APIs :-)

**Comment by dshokouhi** [comment]
Part of the reason why we went with Play Services is that its very battery friendly and also very easy to implement, it is not just a simple switch of the API. 

Google does indeed recommend using play services API in this case as we can see from this link: 

https://developer.android.com/reference/android/location/package-summary

We are open source so anyone who wants to develop this feature is more than welcome to.

**Comment by answer35** [comment]
I am also using a De-googled android and I have to admit that if a solution can be found to get location working it would be amazing. It is really disappointing to be google dependant for a local home solution even if I understand also the reason. Just hoping that someone will be able to help on that part :)

**Comment by gdt** [comment]
Off topic for the issue, but on a ROM with microG, one can run the full version of HAC and get location reported.   You don't need GDR/CloudMessaging enabled.     This still has proprietary code in HAC, but it's talking to the API implemented by microG, and no location is sent to google as far as I know -- the standard belief is that google tracking happens inside play services.  So I think this is safe.    (It's definitely not safe to use a ROM with real play services and have any app have location in the background.)

**Comment by answer35** [comment]
> Off topic for the issue, but on a ROM with microG, one can run the full version of HAC and get location reported. You don't need GDR/CloudMessaging enabled. This still has proprietary code in HAC, but it's talking to the API implemented by microG, and no location is sent to google as far as I know -- the standard belief is that google tracking happens inside play services. So I think this is safe. (It's definitely not safe to use a ROM with real play services and have any app have location in the background.)

I am using a ROM with MicroG but still HA doesn't receive my location. Did you managed to have it working ?

**Comment by answer35** [comment]
I have full version also but still location not working. But I haven't find a way to get precise location in HAC ?!

**Comment by mike-lloyd03** [comment]
@answer35 I am on a microG ROM as well (CalyxOS) and found that unless I enable the "Single Accurate Location" sensor, I will get no location reporting. 

@gdt Is Home Assistant constantly using your location in the background? My green dot in the top right corner is on all day and it's always Home Assistant. I've opened an issue for this (#2695) but it sounds like this issue won't be supported since I'm not on a Google Android ROM. 

I agree with your reasoning that the self-hosted community is also likely to have a number of de-Googled Android users in their midst but I understand that the developers already have a lot on their plate. I guess I need to learn Kotlin.

**Comment by answer35** [comment]
Hmm ok I have found where to set this sensor so location start to work, thank you guys. Now I will check how much battery is getting drained and play with those to adjust it for my needs

**Comment by gdt** [comment]
Yes, the green dot is ~always on.   I think the issue is that CalyxOS is much more aggressive about telling you that the location is in use.  Really you have told HAC to keep track of where you are and report it to your HA, which is of course an ok  tihing to do.  The whole "location notification" scheme is built around the assumption that you have a bunch of sketchy apps that you don't really trust but you want to use them anyway, and it doesn't have support for labeling them trusted vs not trusted.  But that's a CalyxOS lack of feature, not a HAC bug.

**Comment by dshokouhi** [comment]
> I guess I need to learn Kotlin.

If you need any help or have any questions about the app architecture etc.. please join us on discord in #devs_mobile_apps we would be happy to help :)

The green dot feature I see being mentioned is very odd but if its to mean about an app that has an open request for location updates then yes I would expect to see that up all the time but it doesnt mean the app is actively getting updates that very second.  You guys can look at the code and see how we request updates for normal updates and single accurate updates.  On a stock Pixel there is no green dot for any location requests so its definitely added by that OS, I only see it for microphone and camera and only during active use. Have you guys reached out to ask them to turn that off or report a bug in how it reports? That would be a good place to start.

**Comment by dshokouhi** [comment]
It's already mentioned in the companion app docs and project readme.

**Comment by dshokouhi** [comment]
https://companion.home-assistant.io/docs/core/android-flavors

https://github.com/home-assistant/android#app-flavors

It's also mentioned in the f droid installation page

https://f-droid.org/en/packages/io.homeassistant.companion.android.minimal/

**Comment by mike-lloyd03** [comment]
Okay I discovered today that normal, Googled Android hides the location notification from users. I wasn't aware of this. CalyxOS simply enables this feature. It can be toggled in developer settings but this is a system wide setting and cannot be toggled for trusted apps like HAC. 

That being said, I still think there is something unique with how Home Assistant is getting my location. I have OwnTracks running in the background as well and I rarely get any location notifications from it. From the HAC logs, it appears location is updating every few seconds.

**Comment by dshokouhi** [comment]
> From the HAC logs, it appears location is updating every few seconds.

As you are using the single accurate location sensor which you mentioned up above I assume you also needed to enable the setting update with sensors? If so then it would be difficult to predict when it updates as the request is tied to other sensors triggering an update. So indeed it could be updating that frequently based on your own usage. That's not how that sensor was designed to be used. Normal background updates from Google play certified devices get updates once per minute and are not a gps lock.

If you run a custom ROM there is no support for location services, that is why the minimal version exists.

**Comment by mike-lloyd03** [comment]
Yes I suspect it's something to do with MicroG. I'd like to understand more about how this works so I can push for a fix on the MicroG side if needed. 

Could you help me understand how other sensors could be triggering a location update in HAC?

@dshokouhi I understand that I'm working on an unsupported device so I really appreciate the input you've provided so far.

**Comment by dshokouhi** [comment]
> Yes I suspect it's something to do with MicroG. I'd like to understand more about how this works so I can push for a fix on the MicroG side if needed.

FWIW on my amazon tablet I was able to sideload the play store and then installed google maps (opened and granted permissions) to get the device tracker updating. I simply followed the XDA instructions for that. More than likely MicoG is missing more important libraries and APIs.

> Could you help me understand how other sensors could be triggering a location update in HAC?

if you use this option under the single accurate location sensor (Settings > Companion App > Manage Sensors > Single Accurate Location):

![image](https://user-images.githubusercontent.com/1634145/194143915-f50bbfb8-b116-4384-8034-85a9d1180939.png)

then any sensor (and its update frequency) will trigger a GPS location update

https://companion.home-assistant.io/docs/core/sensors#how-sensors-update

**Comment by mike-lloyd03** [comment]
Yes I had "Include in sensor update" enabled. I just disabled it but it still appears that HA is always using location. I see there have been other HA users reporting similar issues in the MicroG repo as well. 

https://github.com/microg/GmsCore/issues/1616

**Comment by dshokouhi** [comment]
> Yes I had "Include in sensor update" enabled. I just disabled it but it still appears that HA is always using location.

May need to wait for all the requests to finish as it doesn't cancel when you turn that off. Also if requests are backed up that may take longer.

I think there is a reason why Google did not enable that green light on their devices, they dont even mention it in official android documentation. It probably doesn't work well. Honestly when dealing with custom ROMs there are too many factors to consider, this is why Google has a certification process to make sure things work as expected. Even if MicroG manages to fix that issue whose to say it wont break in the next dependency update or Android OS update?

**Comment by mike-lloyd03** [comment]
I'm still not sure this is totally a MicroG issue since this problem doesn't happen with any other apps (including OwnTracks which has the same permissions enabled). 

With Home Assistant aiming to be a privacy first application and seeking to help users to be free from big-tech vendor lock-in, I think it's reasonable that a good portion of HA's users will be in a similar situation as those of us here on custom ROMs.

I'll continue to work on this from both ends and see if we can find a resolution.

**Comment by gdt** [comment]
The real issue with the location indicator is that there are apps that do access location more or less continuously, and HA Companion is one of them.  That's legitimate; the user asked for it.   But because we have a culture of people running sketchy apps and then after the fact trying to sort of monitor and contain them, we have these mechanisms.   The green dot really needs a toggle per app to be excluded.   Basically, I know HA is accessing location because I told it to, and I'm ok with that.  It would be nice to know if there was a different location access that was not expected.  But telling me my phone is doing what I asked is not helpful.

So I think the "green light is on" is totally out of scope for HA Companion (and I'm a CalyxOS user that doesn't want to run proprietary software).

**Comment by mike-lloyd03** [comment]
@gdt That very well may be. A per-app toggle in MicroG would be very helpful. 

Do you have any idea why OwnTracks doesn't seem to have this issue though?

**Comment by ale5000-git** [comment]
microG can be used also in place of Google libraries to compile apps.
I'm not sure if it contains everything need for your app but have anyone tried to compile the full version with microG libs?

**Comment by guttermonk** [comment]
I'm also using the "Home Assistant Minimal" version of the companion app from F-droid on GrapheneOS and could benefit from the addition of location sensors. I agree with some of the comments above that users that want to roll their own home automation system should have the option for it to be as de-Googled as one chooses without losing functionality. It would be nice to see feature parity between the F-droid and Google Play versions of the app. I've been on the Home Assistant journey for over 4 years, and look forward to the continued development.

**Comment by Forage** [comment]
Not to continue the off-topic MicroG discussion, but just a quick heads-up for those with MicroG background location reporting issues, that's fixed when updating to a more recent version of MicroG: https://github.com/microg/GmsCore/issues/1616#issuecomment-2061736257.

**Comment by i00** [comment]
I want this too! +1

**Comment by kuba-orlik** [comment]
I'm trying to understand why adding a google library is necessary for the companion app to access location? I know many applications that access location data and are available on F-Droid without any apparent restrictions :thinking:

**Comment by gdt** [comment]
It's not necessary.   Basically, with AOSP interfaces, you can get GPS location and you can get network location.   Google libraries in proprietary android have  a fused interface that is apparently easier to use, and that's how the app was written.   So someone just needs to write an alternate implementation to be used when building w/o proprietary libs.

**Comment by emailforos** [comment]
Well, another one with location problem in the android app.

I'm a Huawei phone user so everybody knows We're not able to use any of the Google Apps. However, I've got several apps using geolocation and they work perfectly.

I am not a programmer, I am a grateful user of Home Assistant, and I understand that it is complicated to cover all the problems that come out of HA. But if HA stands out in anything, it is the search for independence from proprietary software/apis and what more independence than to stop depending on Google's APIs. 

Thanking the work of all HA staff.

**Comment by licaon-kter** [comment]
So what's the blocker here? There 're plenty of apps in F-Droid that can use location just fine, in all sort of languages.

**Comment by moritzruth** [comment]
And also https://github.com/owntracks/android (see [Build flavours](https://github.com/owntracks/android?tab=readme-ov-file#build-flavours)).

**Comment by licaon-kter** [comment]
@TimoPtr what's "location tracking"? OSMAnd~ and OrganicMaps work fine.

OrganicMaps even has microG integration so on Google Services enabled devices it can use that. All FOSS.

**Comment by TimoPtr** [comment]
So I've been playing around with the AOSP Location API instead of GMS for the GeocodeSensor. I would need some people to test it before merging it. If it works I might consider to do the same for the rest of the sensors for location.

You can find the artifact inside the Check section of my PR https://github.com/home-assistant/android/pull/5148
Or here
https://github.com/home-assistant/android/actions/runs/14062886260/artifacts/2816647006

**Comment by dallyger** [comment]
@TimoPtr how can I help to test? I've installed the minimal app APK from your linked zip and from the PR artifacts, however there are no location sensors available. Tried both versions: `2025.3.5-beta.0.14+af7e384-minimal` and `2025.3.5-beta.0.17+52e645b-minimal`. Do I have to install the `full` build?

**Comment by TimoPtr** [comment]
> [@TimoPtr](https://github.com/TimoPtr) how can I help to test? I've installed the minimal app APK from your linked zip and from the PR artifacts, however there are no location sensors available. Tried both versions: `2025.3.5-beta.0.14+af7e384-minimal` and `2025.3.5-beta.0.17+52e645b-minimal`. Do I have to install the `full` build?

You did right. You should look for the geocode sensor not location.

**Comment by dallyger** [comment]
Can confirm, that the geocode sensor has data on my Nothing Phone 1 (Nothing OS 3.0, based on Android 15). However, on my OnePlus 6 with LineageOS 22.1 (Android 15 too, but without microG), the sensor is not even available in the list. Is there some way to force displaying it? Or is microG a requirement?

**Comment by TimoPtr** [comment]
microG is not a requirements. This PR is based on pure AOSP API. The only thing that could make the difference if you used the same APK is 
```
    override fun hasSensor(context: Context): Boolean {
        return Geocoder.isPresent()
    }
```

Here the documentation about this function
```
    /**
     * Returns true if there is a geocoder implementation present on the device that may return
     * results. If true, there is still no guarantee that any individual geocoding attempt will
     * succeed.
     */
```

It means that maybe the geocoder is not implemented for your phone on LineageOS.

**Comment by wabuMike** [comment]
> It means that maybe the geocoder is not implemented for your phone on LineageOS.

@TimoPtr Installed the build and can confirm that the 'geocode' sensor is not available on a Fairphone 3 running with LineageOS. Probably not implemented.

**Comment by TimoPtr** [comment]
I cannot commit for when exactly but I'm going to implement another sensor without GMS so that you can test it.

**Comment by nabeelr** [comment]
Throwing my hat into the ring on this one. Getting location on the F-Droid version would be great.

**Comment by mboutolleau** [comment]
I just tested the minimal APK in #5148 and the Geocoded location sensor works perfectly on a Fairphone 3 with the official ROM (Android 13, 2025-03-05 security patch).
Thank you @TimoPtr

**Comment by jahknem** [comment]
I tested the full APK in #5148 on a Pixel 9 Pro and the geocoded location sensor also works perfectly on BP1A.250405.007.C1
I get an error about the firebase API not working which will inhibit notifications, but I presume that was to be expected?
In any case, thank you @TimoPtr , this is a gods sent.

**Comment by TimoPtr** [comment]
The PR has been merged 🚀

**Comment by Pawik55** [comment]
Tested the release build on my Android 8 phone and it does not work. Looking at the log it fails at the check for the GPS location timestamp for being to old: https://github.com/home-assistant/android/blob/b7281155f1bdc0bf53780e8b304cda1ae0532fd4/app/src/main/kotlin/io/homeassistant/companion/android/sensors/GeocodeSensorManager.kt#L102

It seems to be the exact same issue as described in this stackoverflow post: https://stackoverflow.com/questions/74126739/getting-wrong-and-very-old-timestamps-from-location-gettime

**Comment by dshokouhi** [comment]
> Looking at the log it fails

please paste the error log

**Comment by gdt** [comment]
> Tested the release build on my Android 8 phone and it does not work. Looking at the log it fails at the check for the GPS location timestamp for being to old:
> 
> [android/app/src/main/kotlin/io/homeassistant/companion/android/sensors/GeocodeSensorManager.kt](https://github.com/home-assistant/android/blob/b7281155f1bdc0bf53780e8b304cda1ae0532fd4/app/src/main/kotlin/io/homeassistant/companion/android/sensors/GeocodeSensorManager.kt#L102)
> 
> Line 102 in [b728115](/home-assistant/android/commit/b7281155f1bdc0bf53780e8b304cda1ae0532fd4)
>  Timber.w("Skipping geocoded update due to old timestamp ${location.instant()}") 
> 
> It seems to be the exact same issue as described in this stackoverflow post: https://stackoverflow.com/questions/74126739/getting-wrong-and-very-old-timestamps-from-location-gettime

GPS transmits a 10-bit week number and then time within week.  (Later signals have 13 bits, I think, but many receivers only decode the 10-bit kind.)   Receivers have to figure out the rest.   10 bits of week is about 19 years; "W1024" was a big deal in IIRC August  1999 (from 1980), and W2048 in 2019 or so.    If the time is ~exactly 7*1024 days early, but the right time, that's almost certainly what's going on.    if so, this is basically a bug in your device or "device firmware is too old".

Sometimes code has workarounds to check the timestamp for being very old, and adding 7*1024 days if that works to get it close to the believed time according to other sources.  Basically, if GPS timestamp is more than 19 years before now, then add 1024 weeks, and repeat.

Alternatively, use the location and ignore the time.

**Comment by Pawik55** [comment]
This seems to be the exact issue, but with the exception that it worked for one day. Maybe this depends on the satellites the device gets its position.

Here line of the log with the warning. I don't think the rest is necessary.
```
06-11 12:01:35.886 13976 13976 I InputMethodManager: startInputInner - mService.startInputOrWindowGainedFocus
06-11 12:01:35.913 13976 17851 W GeocodeSensorManager: Skipping geocoded update due to old timestamp 2005-10-26T10:01:37.437Z
06-11 12:01:35.915 13976 21202 D OpenGLRenderer: eglDestroySurface = 0x7781ce3eb0
```

**Comment by Pawik55** [comment]
@TimoPtr There is still a small bug with the rollover present. Sometimes or probably more often than not the time of the gps timestamp lies a few seconds ahead of the time of the device. You can even see this in my post above this post. This line is the problem:
https://github.com/home-assistant/android/blob/main/app/src/main/kotlin/io/homeassistant/companion/android/sensors/GeocodeSensorManager.kt#L191

It's a few seconds below 1024 weeks thats why the check won't detect it correctly. 
Would you change it to something like this:
`return if (this >= GPS_ROLLOVER_WEEKS1024 - 5.minutes) {`
I can also make another merge request if you don't want to.

@gdt I just wanted to say thank you. Because of you the bug could be fixed.

**Comment by TimoPtr** [comment]
> @TimoPtr There is still a small bug with the rollover present. Sometimes or probably more often than not the time of the gps timestamp lies a few seconds ahead of the time of the device. You can even see this in my post above this post. This line is the problem:
> https://github.com/home-assistant/android/blob/main/app/src/main/kotlin/io/homeassistant/companion/android/sensors/GeocodeSensorManager.kt#L191
> 
> It's a few seconds below 1024 weeks thats why the check won't detect it correctly. 
> Would you change it to something like this:
> `return if (this >= GPS_ROLLOVER_WEEKS1024 - 5.minutes) {`
> I can also make another merge request if you don't want to.
> 
> @gdt I just wanted to say thank you. Because of you the bug could be fixed.

Feel free to open a PR with your changes and add proper tests to showcase what your proposal is fixing.

**Comment by user-mb** [comment]
i tested minimal flavor on GrapheneOS and cant seem to find the Geocoded location sensor. am i missing something?

i think it is not implemented:
https://github.com/GrapheneOS/os-issue-tracker/issues/2070#issuecomment-1492868238

**Comment by Eoghanmc22** [comment]
@user-mb Geolocation is implemented in GOS on an opt-in basis as of version [2025080400](https://grapheneos.org/releases#2025080400) and I was able to get the Geocoded location sensor working on latest GOS. However I was disappointed to find out that my device did not appear on the home assistant map since the device tracker sensor does not appear to be implemented in the minimal build of the companion app.

Is there anything blocking a device tracker sensor implementation based on AOSP apis?

**Comment by gdt** [comment]
WIthout reading the code, it seems obvious that given the use of the non-AOSP Google Fused Location API, that when the "minimal" version (which should be really be thought of as producing a Free Software binary apk, not reduced functionality), then the code using Fused Location had to be omitted.  And almost certainly, code that depended on that API was just included in the grand #ifdef.   Really there should be an abstract class within HAC and then a Google proprietary implementation and a AOSP-based implementation.  The location sensor code shouldn't care; it just gets location updates.     Thus it seems likely a small matter of programming to adjust this.

**Comment by TimoPtr** [comment]
> WIthout reading the code, it seems obvious that given the use of the non-AOSP Google Fused Location API, that when the "minimal" version (which should be really be thought of as producing a Free Software binary apk, not reduced functionality), then the code using Fused Location had to be omitted.  And almost certainly, code that depended on that API was just included in the grand #ifdef.   Really there should be an abstract class within HAC and then a Google proprietary implementation and a AOSP-based implementation.  The location sensor code shouldn't care; it just gets location updates.     Thus it seems likely a small matter of programming to adjust this.

I did a first implementation of the geocoder sensor since it didn't require a lot of effort but for the rest it needs a quite big refactor https://github.com/home-assistant/android/issues/1443
Which I'm not planning to take in the next few months, since I'm focused on other big topics. 
At some I'll come to this for sure. But if someone is willing to help on this I'm willing to help 👍 in the design and review. 

The plan is to have a location api agnostic implementation of the locationManager.

**Comment by mgolik** [comment]
I am using F-droid version on Samsung phone. My "Geocoded location" is working properly - locations names are OK and attributes contain GPS coordinates. But "device_tracker" is in unknown state with one attribute "source" and value "GPS".
Is this expected? I can't use "Geocoded location" on map.

I also installed Play store version where I have "Geocoded location" sensor disabled, yet "device_tracker" entity contains GPS data.

**Comment by DerRockWolf** [comment]
@mgolik correct currently only the geocoded locations sensor is implemented in the FOSS version.
See the last few messages in this issue.

**Comment by mgolik** [comment]
Thanks for clarification! I was not sure if / how the sensors are related but now I know for sure ;)

**Comment by sangvikh** [comment]
I found a solution for the missing device tracker - I set up an automation that creates a virtual device tracker from the geolocation sensor:

````
alias: Person tracker from sm_a155f geocoded location
description: ""
triggers:
  - trigger: state
    entity_id:
      - sensor.sm_a155f_geocoded_location
conditions:
  - condition: template
    value_template: >-
      {{ state_attr('sensor.sm_a155f_geocoded_location', 'location') is not none
      }}
actions:
  - action: device_tracker.see
    metadata: {}
    data:
      dev_id: sm_a155f_virtual
      gps: "{{ state_attr('sensor.sm_a155f_geocoded_location', 'location') }}"
      gps_accuracy: 50
mode: single
````

**Comment by TimoPtr** [comment]
> I found a solution for the missing device tracker - I set up an automation that creates a virtual device tracker from the geolocation sensor:
> 
> ```
> alias: Person tracker from sm_a155f geocoded location
> description: ""
> triggers:
>   - trigger: state
>     entity_id:
>       - sensor.sm_a155f_geocoded_location
> conditions:
>   - condition: template
>     value_template: >-
>       {{ state_attr('sensor.sm_a155f_geocoded_location', 'location') is not none
>       }}
> actions:
>   - action: device_tracker.see
>     metadata: {}
>     data:
>       dev_id: sm_a155f_virtual
>       gps: "{{ state_attr('sensor.sm_a155f_geocoded_location', 'location') }}"
>       gps_accuracy: 50
> mode: single
> ```

Interesting that could be a blueprint that you could share on the https://companion.home-assistant.io/ for people using the minimal flavor and looking for a solution. Thanks for sharing.

**Comment by Soukyuu** [comment]
Which version has the geocoded location sensor? I have the 2025.11.1-minimal installed and even though android says it's making use of the location permission (last time at 12:00 pm), I don't see any location sensors available in the app... 

edit: nvm, I didn't see the posts github had collapsed, and one of them contained the hint: you have to explicitly enable geocoder on GrapheneOS

**Comment by sangvikh** [comment]
> Which version has the geocoded location sensor? I have the 2025.11.1-minimal installed and even though android says it's making use of the location permission (last time at 12:00 pm), I don't see any location sensors available in the app...
> 
> edit: nvm, I didn't see the posts github had collapsed, and one of them contained the hint: you have to explicitly enable geocoder on GrapheneOS

I am using 2025.11.1-minimal.
The sensor must be enabled in Settings -> Companion app -> Manage sensors. Location permission must also be set in the phone settings.

**Comment by Soukyuu** [comment]
It's working now, had to also enable the Geocoder on GrapheneOS. Used Owntracks before, but it can only push to one HTTP endpoint and I have two servers, so I didn't want to setup an MQTT server just for that. 

The geocoding sensor is a very welcome option until someone finds time to refactor the LocationManager.

**Comment by Soukyuu** [comment]
Ok, something broke. My geocoded location is stuck for several days now, while owntracks is reporting correctly. Nothing I tried (restarting app, re-enabling sensor, changing refresh interval) helps. I'm on 2025.11.3-minimal now. How can I debug this?
